packages/remix-react/routes.tsx

```diff
@@ -78,6 +80,42 @@ export interface ClientRoute extends Route {
 
 type RemixRouteComponentType = ComponentType<{ id: string }>;
 
+export function createServerRoutes(
+  manifest: RouteManifest<EntryRoute>,
+  routeModules: RouteModules,
+  parentId?: string
+): DataRouteObject[] {
+  return Object.values(manifest)
+    .filter((route) => route.parentId === parentId)
+    .map((route) => {
+      let baseRoute: Omit<DataRouteObject, "children" | "index"> = {
+        caseSensitive: route.caseSensitive,
+        element: <RemixRoute id={route.id} />,
+        errorElement:
+          route.id === "root" ||
+          routeModules[route.id].ErrorBoundary != null ||
+          routeModules[route.id].CatchBoundary != null ? (
+            <RemixRouteError id={route.id} />
+          ) : undefined,
+        id: route.id,
+        path: route.path,
+        // Note: we don't need loader/action/shouldRevalidate on these routes
+        // since they're for a static render
+        handle: routeModules[route.id].handle,
+      };
```

Comment:
Server-render-time routes don't have loaders/actions since we already fetched the data!